In [1]:
import pandas as pd

In [2]:
for i in range(1,25):
    z = str(i)
    file_name = 'File' + ' ' + z + '.csv'
    globals()['table%s' % z] = pd.read_csv("ifct-composition-assets-renamed/%s" %file_name)

In [3]:
for i in range(1,5):
    z = str(i)
    globals()['table%s_merged' % z] = globals()['table%s' % z]

In [4]:
table5_merged = pd.merge(table5,table6)
table6_merged = table7
#table7_list = [table8,table9,table10,table11,table12,table13,table14]
#table7_merged = pd.concat(table7_list)

# Do we need a fix here??
for i in range(8,10):
    z = str(i)
    table7_merged = pd.merge(globals()['table%s' % z],globals()['table%s' % str(i+1)],on='code',how='outer')

In [5]:
table7

code                                             name  regn         cho  \
0     A001       Amaranth seed, black (Amaranthus cruentus)     1       56.71   
1     A002  Amaranth seed, pale brown (Amaranthus cruentus)     6  60.13±1.30   
2     A003                    Bajra (Pennisetum typhoideum)     6  56.02±2.57   
3     A004                         Barley (Hordeum vulgare)     6  72.73±2.93   
4     A005                          Jowar (Sorghum vulgare)     6  60.96±1.70   
..     ...                                              ...   ...         ...   
309   K002                                    Coconut Water     6   3.06±0.07   
310  L001                              Milk, whole, Buffalo     6  5.48±0.19    
311   L002                                 Milk, whole, Cow     6   4.94±0.18   
312   L003                                           Paneer     6  12.01±0.08   
313   L004                                             Khoa     6  16.33±3.84   

         starch        frus        glus        sucs mals      fsugar  \
0         55.83         0.1        0.22        0.46  0.1        0.88   
1    59.33±1.31   0.10±0.01   0.22±0.04   0.48±0.04  NaN   0.80±0.07   
2    55.21±2.57   0.21±0.01   0.60±0.02         NaN  NaN   0.81±0.01   
3    72.67±2.92         NaN   0.06±0.02         NaN  NaN   0.06±0.02   
4    59.70±1.70   0.57±0.07   0.10±0.01   0.60±0.04  NaN   1.27±0.05   
..          ...         ...         ...         ...  ...         ...   
309         NaN   3.06±0.07         NaN         NaN  NaN   3.06±0.07   
310         NaN  0.24±0.05   0.39±0.04   0.59±0.06   NaN   5.34±0.16   
311         NaN   0.20±0.04   0.18±0.04   0.24±0.03  NaN   4.89±0.14   
312         NaN   0.50±0.03   1.11±0.24   0.84±0.08  NaN  11.57±0.28   
313         NaN   0.80±0.04   1.57±0.34   0.91±0.04  NaN  16.14±1.11   

        lactose  
0           NaN  
1           NaN  
2           NaN  
3           NaN  
4           NaN  
..          ...  
309         NaN  
310   4.12±0.14  
311   4.28±0.10  
312   9.12±0.20  
313  12.86±1.03  

[314 rows x 11 columns]

In [6]:
#ignoring table 7 and onwards as they contain info that is not relevant on the macro scale

In [7]:
table7_merged

code                                           name_x  regn_x  \
0    A001       Amaranth seed, black (Amaranthus cruentus)     1.0   
1    A002  Amaranth seed, pale brown (Amaranthus cruentus)     6.0   
2    A003                    Bajra (Pennisetum typhoideum)     6.0   
3    A004                         Barley (Hordeum vulgare)     6.0   
4    A005                          Jowar (Sorghum vulgare)     6.0   
..    ...                                              ...     ...   
406  O059                                              NaN     NaN   
407  O060                                              NaN     NaN   
408  O061                                              NaN     NaN   
409  O062                                              NaN     NaN   
410  O063                                              NaN     NaN   

       f20d1n9f f22d1n9 f24d1c   f18d2cn6 f20d2     f18d3n3 f20d4n6  ...  \
0         12.62     NaN    NaN       2259   NaN       20.91     NaN  ...   
1    12.71±1.56     NaN    NaN   2223±129   NaN  43.03±2.70     NaN  ...   
2           NaN     NaN    NaN  1844±56.7   NaN     140±5.8     NaN  ...   
3           NaN     NaN    NaN   549±21.6   NaN  26.32±3.35     NaN  ...   
4           NaN     NaN    NaN   508±18.3   NaN  16.54±1.31     NaN  ...   
..          ...     ...    ...        ...   ...         ...     ...  ...   
406         NaN     NaN    NaN        NaN   NaN         NaN     NaN  ...   
407         NaN     NaN    NaN        NaN   NaN         NaN     NaN  ...   
408         NaN     NaN    NaN        NaN   NaN         NaN     NaN  ...   
409         NaN     NaN    NaN        NaN   NaN         NaN     NaN  ...   
410         NaN     NaN    NaN        NaN   NaN         NaN     NaN  ...   

    regn_y f11d0 f12d0  f14d0  f15d0  f16d0 f18d0 f20d0  f22d0 f24d0  
0      NaN   NaN   NaN    NaN    NaN    NaN   NaN   NaN    NaN   NaN  
1      NaN   NaN   NaN    NaN    NaN    NaN   NaN   NaN    NaN   NaN  
2      NaN   NaN   NaN    NaN    NaN    NaN   NaN   NaN    NaN   NaN  
3      NaN   NaN   NaN    NaN    NaN    NaN   NaN   NaN    NaN   NaN  
4      NaN   NaN   NaN    NaN    NaN    NaN   NaN   NaN    NaN   NaN  
..     ...   ...   ...    ...    ...    ...   ...   ...    ...   ...  
406    1.0   NaN   NaN  80.96    NaN    827   121   NaN    NaN   NaN  
407    1.0   NaN   NaN  59.34    NaN    754    97   NaN    NaN   NaN  
408    1.0   NaN   NaN    282    NaN   2634   227   NaN    NaN   NaN  
409    1.0   NaN   NaN    180    NaN   1606   121   NaN    NaN   NaN  
410    1.0   NaN   NaN    163    NaN   1295   117   NaN    NaN   NaN  

[411 rows x 24 columns]

In [8]:
table_dict = {'table1_merged':'proximate principles and dietary fibre','table2_merged':'water soluble vitamins','table3_merged':'fat soluble vitamins','table4_merged':'carotenoids','table5_merged':'minerals and trace elements','table6_merged':'starch and individual sugars'}

In [9]:
for i in range(1,6):
    z = str(i)
    table_merged_master = pd.merge(globals()['table%s_merged' % z],globals()['table%s_merged' % str(i+1)],on='code',how='outer')

In [10]:
table_merged_master

code                                           name_x  regn_x  \
0     A001       Amaranth seed, black (Amaranthus cruentus)     1.0   
1     A002  Amaranth seed, pale brown (Amaranthus cruentus)     6.0   
2     A003                    Bajra (Pennisetum typhoideum)     6.0   
3     A004                         Barley (Hordeum vulgare)     6.0   
4     A005                          Jowar (Sorghum vulgare)     6.0   
..     ...                                              ...     ...   
525   E063                                              NaN     NaN   
526   F001                                              NaN     NaN   
527  J001                                               NaN     NaN   
528  K001                                               NaN     NaN   
529  L001                                               NaN     NaN   

            al         as           cd          ca           cr           co  \
0         3.32        NaN          NaN         181        1.227        0.059   
1    2.73±0.47        NaN  0.001±0.000    162±15.7  0.092±0.045  0.021±0.005   
2    2.21±0.78  0.97±0.24  0.003±0.001  27.35±2.16  0.025±0.006  0.030±0.015   
3          NaN        NaN          NaN  28.64±3.49  0.029±0.009  0.027±0.010   
4    2.56±0.59  1.53±0.04  0.002±0.002  27.60±3.71  0.010±0.003  0.012±0.007   
..         ...        ...          ...         ...          ...          ...   
525        NaN        NaN          NaN         NaN          NaN          NaN   
526        NaN        NaN          NaN         NaN          NaN          NaN   
527        NaN        NaN          NaN         NaN          NaN          NaN   
528        NaN        NaN          NaN         NaN          NaN          NaN   
529        NaN        NaN          NaN         NaN          NaN          NaN   

            cu  ...                                           name_y regn_y  \
0         0.81  ...       Amaranth seed, black (Amaranthus cruentus)    1.0   
1    0.56±0.09  ...  Amaranth seed, pale brown (Amaranthus cruentus)    6.0   
2    0.54±0.11  ...                    Bajra (Pennisetum typhoideum)    6.0   
3    0.43±0.17  ...                         Barley (Hordeum vulgare)    6.0   
4    0.45±0.11  ...                          Jowar (Sorghum vulgare)    6.0   
..         ...  ...                                              ...    ...   
525        NaN  ...                   Strawberry (Fragaria ananassa)    6.0   
526        NaN  ...                        Beet root (Beta vulgaris)    6.0   
527        NaN  ...           Button mushroom, fresh (Agaricus sp.)     1.0   
528        NaN  ...                                           Toddy    10.0   
529        NaN  ...                             Milk, whole, Buffalo    6.0   

            cho      starch        frus        glus        sucs mals  \
0         56.71       55.83         0.1        0.22        0.46  0.1   
1    60.13±1.30  59.33±1.31   0.10±0.01   0.22±0.04   0.48±0.04  NaN   
2    56.02±2.57  55.21±2.57   0.21±0.01   0.60±0.02         NaN  NaN   
3    72.73±2.93  72.67±2.92         NaN   0.06±0.02         NaN  NaN   
4    60.96±1.70  59.70±1.70   0.57±0.07   0.10±0.01   0.60±0.04  NaN   
..          ...         ...         ...         ...         ...  ...   
525   3.66±0.11   0.10±0.01   1.90±0.05   1.60±0.14   0.07±0.03  NaN   
526   6.04±0.94   1.69±0.43   1.67±0.02   1.46±0.02   1.21±0.03  NaN   
527         0.8         0.7         0.1         NaN         NaN  NaN   
528  5.73±0.17          NaN  5.73±0.17          NaN         NaN  NaN   
529  5.48±0.19          NaN  0.24±0.05   0.39±0.04   0.59±0.06   NaN   

        fsugar    lactose  
0         0.88        NaN  
1    0.80±0.07        NaN  
2    0.81±0.01        NaN  
3    0.06±0.02        NaN  
4    1.27±0.05        NaN  
..         ...        ...  
525  3.56±0.11        NaN  
526  4.35±0.02        NaN  
527        0.1        NaN  
528  5.73±0.17        NaN  
529  5.34±0.16  4.12±0.14  

[530 rows x 33 columns]

---

## Filtering Recipes

In [11]:
# Importing dataset into the workspace
globals()['IndianRecipes'] = pd.read_csv("IndianFoodDataset/IndianFoodDatasetCSV.csv")
# Displaying
IndianRecipes

Srno                                         RecipeName  \
0         1                               Masala Karela Recipe   
1         2  टमाटर पुलियोगरे रेसिपी - Spicy Tomato Rice (Re...   
2         3  Ragi Semiya Upma Recipe - Ragi Millet Vermicel...   
3         4  Gongura Chicken Curry Recipe - Andhra Style Go...   
4         5  आंध्रा स्टाइल आलम पचड़ी रेसिपी - Adrak Chutney ...   
...     ...                                                ...   
6866  14073  गोअन मशरुम जकुटी रेसिपी - Goan Mushroom Xacuti...   
6867  14107  शकरकंदी और मेथी का पराठा रेसिपी - Sweet Potato...   
6868  14165      Ullikadala Pulusu Recipe | Spring Onion Curry   
6869  14167  Kashmiri Style Kokur Yakhni Recipe-Chicken Coo...   
6870  14211              नवरंग दाल रेसिपी - Navrang Dal Recipe   

                                   TranslatedRecipeName  \
0                                  Masala Karela Recipe   
1                            Spicy Tomato Rice (Recipe)   
2     Ragi Semiya Upma Recipe - Ragi Millet Vermicel...   
3     Gongura Chicken Curry Recipe - Andhra Style Go...   
4     Andhra Style Alam Pachadi Recipe - Adrak Chutn...   
...                                                 ...   
6866                        Goan Mushroom Xacuti Recipe   
6867        Sweet Potato & Methi Stuffed Paratha Recipe   
6868      Ullikadala Pulusu Recipe | Spring Onion Curry   
6869  Kashmiri Style Kokur Yakhni Recipe-Chicken Coo...   
6870                                 Navrang Dal Recipe   

                                            Ingredients  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...   
1     2-1/2 कप चावल - पका ले,3 टमाटर,3 छोटा चमच्च बी...   
2     1-1/2 cups Rice Vermicelli Noodles (Thin),1 On...   
3     500 grams Chicken,2 Onion - chopped,1 Tomato -...   
4     1 बड़ा चमच्च चना दाल,1 बड़ा चमच्च सफ़ेद उरद दाल,2...   
...                                                 ...   
6866  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...   
6867  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...   
6868  150 grams Spring Onion (Bulb & Greens) - chopp...   
6869  1 kg Chicken - medium pieces,1/2 cup Mustard o...   
6870  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...   

                                  TranslatedIngredients  PrepTimeInMins  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...              15   
1     2-1 / 2 cups rice - cooked, 3 tomatoes, 3 teas...               5   
2     1-1/2 cups Rice Vermicelli Noodles (Thin),1 On...              20   
3     500 grams Chicken,2 Onion - chopped,1 Tomato -...              15   
4     1 tablespoon chana dal, 1 tablespoon white ura...              10   
...                                                 ...             ...   
6866  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...              15   
6867  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...              30   
6868  150 grams Spring Onion (Bulb & Greens) - chopp...               5   
6869  1 kg Chicken - medium pieces,1/2 cup Mustard o...              30   
6870  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...              10   

      CookTimeInMins  TotalTimeInMins  Servings               Cuisine  \
0                 30               45         6                Indian   
1                 10               15         3  South Indian Recipes   
2                 30               50         4  South Indian Recipes   
3                 30               45         4                Andhra   
4                 20               30         4                Andhra   
...              ...              ...       ...                   ...   
6866              45               60         4          Goan Recipes   
6867              60               90         4  North Indian Recipes   
6868              10               15         2                Andhra   
6869              45               75         4              Kashmiri   
6870              30               40         4  North Indian Recipes   

                  

In [12]:
# Source: https://celiac.org/gluten-free-living/what-is-gluten/sources-of-gluten/
glutenContainingIngredients = ["wheat","wheatberries","durum","emmer","semolina","spelt","farina","farro","graham","rye","barley","triticale","malt","brewer's yeast"]
part2 = ["pasta", "raviolis", "dumplings", "couscous", "gnocchi", "noodles", "ramen", "udon", "soba", "chow mein", "egg noodles", "croissant", "pita", "naan", "bagel", "flatbread", "cornbread", "potato bread", "muffins", "donuts", "rolls", "pretzels", "goldfish", "graham crackers", "cake", "cakes", "cookies", "pie crusts", "brownies", "corn flakes", "rice puffs", "pancakes", "waffles", "french toast", "crepes", "biscuit","biscuits", "panko breadcrumbs", "soy sauce", "flour tortillas", "beer", "bread", "atta", "sooji", "suji", "biscuits"]
glutenContainingIngredients += part2
glutenContainingIngredients += ['whole wheat flour', 'all purpose flour','maida']

glutenContainingIngredients

['wheat',
 'wheatberries',
 'durum',
 'emmer',
 'semolina',
 'spelt',
 'farina',
 'farro',
 'graham',
 'rye',
 'barley',
 'triticale',
 'malt',
 "brewer's yeast",
 'pasta',
 'raviolis',
 'dumplings',
 'couscous',
 'gnocchi',
 'noodles',
 'ramen',
 'udon',
 'soba',
 'chow mein',
 'egg noodles',
 'croissant',
 'pita',
 'naan',
 'bagel',
 'flatbread',
 'cornbread',
 'potato bread',
 'muffins',
 'donuts',
 'rolls',
 'pretzels',
 'goldfish',
 'graham crackers',
 'cake',
 'cakes',
 'cookies',
 'pie crusts',
 'brownies',
 'corn flakes',
 'rice puffs',
 'pancakes',
 'waffles',
 'french toast',
 'crepes',
 'biscuit',
 'biscuits',
 'panko breadcrumbs',
 'soy sauce',
 'flour tortillas',
 'beer',
 'bread',
 'atta',
 'sooji',
 'suji',
 'biscuits',
 'whole wheat flour',
 'all purpose flour',
 'maida']

In [13]:
print(IndianRecipes.describe())

               Srno  PrepTimeInMins  CookTimeInMins  TotalTimeInMins  \
count   6871.000000     6871.000000     6871.000000      6871.000000   
mean    4902.979042       28.585213       30.832339        59.417552   
std     3479.531022       81.042007       34.019694        88.699940   
min        1.000000        0.000000        0.000000         0.000000   
25%     1955.500000       10.000000       20.000000        30.000000   
50%     4304.000000       15.000000       30.000000        40.000000   
75%     7363.000000       20.000000       35.000000        55.000000   
max    14211.000000     2880.000000      900.000000      2925.000000   

          Servings  
count  6871.000000  
mean      5.611410  
std      26.221807  
min       1.000000  
25%       4.000000  
50%       4.000000  
75%       4.000000  
max    1000.000000  


In [14]:
IndianRecipes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6871 entries, 0 to 6870
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Srno                    6871 non-null   int64 
 1   RecipeName              6871 non-null   object
 2   TranslatedRecipeName    6871 non-null   object
 3   Ingredients             6865 non-null   object
 4   TranslatedIngredients   6865 non-null   object
 5   PrepTimeInMins          6871 non-null   int64 
 6   CookTimeInMins          6871 non-null   int64 
 7   TotalTimeInMins         6871 non-null   int64 
 8   Servings                6871 non-null   int64 
 9   Cuisine                 6871 non-null   object
 10  Course                  6871 non-null   object
 11  Diet                    6871 non-null   object
 12  Instructions            6871 non-null   object
 13  TranslatedInstructions  6871 non-null   object
 14  URL                     6871 non-null   object
dtypes: i

In [15]:
f=0
c=0
for x,y,i in zip(IndianRecipes["TranslatedIngredients"],IndianRecipes["Srno"],range(1,6872)):
    ingredientsList = str(x).split(",")
    #print(ingredientsList) # For testing purposes
    for ingredient in ingredientsList:
        for glutenContainingIngredient in glutenContainingIngredients:
            if glutenContainingIngredient.lower() in ingredient.lower():
                print("CONTAINS GLUTEN!",i,y,glutenContainingIngredient,ingredient)
                # For testing purposes
                IndianRecipes.drop(i-1,inplace=True)
                f = 1
                c += 1
                break;
        if f==1:
            break;
    if f==1:
        f=0
        continue;


CONTAINS GLUTEN! 3 3 noodles 1-1/2 cups Rice Vermicelli Noodles (Thin)
CONTAINS GLUTEN! 10 10 rye  1 teaspoon Rye
CONTAINS GLUTEN! 12 12 wheat 1/2 cup Whole Wheat Bread crumbs
CONTAINS GLUTEN! 14 14 atta 2 Bay leaves (tej patta)
CONTAINS GLUTEN! 16 16 rye  as per taste Rye
CONTAINS GLUTEN! 17 17 atta 1 Bay leaf (tej patta)
CONTAINS GLUTEN! 19 19 all purpose flour 2 tablespoon All Purpose Flour (Maida)
CONTAINS GLUTEN! 21 21 rye  1/2 tsp rye
CONTAINS GLUTEN! 22 22 wheat 1/2 cup Whole Wheat Flour
CONTAINS GLUTEN! 23 23 bread 2 Submarine Bread (Subway Bread) - (flat breads or foot longs)
CONTAINS GLUTEN! 27 29 atta 1 Bay leaf (tej patta)
CONTAINS GLUTEN! 28 31 pasta 1 tablespoon Roasted tomato pasta sauce
CONTAINS GLUTEN! 29 32 wheat 1 cup Whole Wheat Flour
CONTAINS GLUTEN! 30 33 soy sauce 1 tablespoon Soy sauce
CONTAINS GLUTEN! 31 34 all purpose flour All Purpose Flour (Maida)
CONTAINS GLUTEN! 33 36 atta 1 cup Cabbage (Patta Gobi/ Muttaikose) - shredded
CONTAINS GLUTEN! 34 37 all purpose

In [16]:
# re-running to check if all gluten containing items have been removed
for x,y,i in zip(IndianRecipes["TranslatedIngredients"],IndianRecipes["Srno"],range(1,6872)):
    ingredientsList = str(x).split(",")
    #print(ingredientsList) # For testing purposes
    for ingredient in ingredientsList:
        for glutenContainingIngredient in glutenContainingIngredients:
            if glutenContainingIngredient.lower() in ingredient.lower():
                print("CONTAINS GLUTEN!",i,y,glutenContainingIngredient,ingredient)
                # For testing purposes
                #IndianRecipes.drop((i-1))

In [17]:
IndianRecipes.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 4512 entries, 0 to 6870
Data columns (total 15 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   Srno                    4512 non-null   int64 
 1   RecipeName              4512 non-null   object
 2   TranslatedRecipeName    4512 non-null   object
 3   Ingredients             4506 non-null   object
 4   TranslatedIngredients   4506 non-null   object
 5   PrepTimeInMins          4512 non-null   int64 
 6   CookTimeInMins          4512 non-null   int64 
 7   TotalTimeInMins         4512 non-null   int64 
 8   Servings                4512 non-null   int64 
 9   Cuisine                 4512 non-null   object
 10  Course                  4512 non-null   object
 11  Diet                    4512 non-null   object
 12  Instructions            4512 non-null   object
 13  TranslatedInstructions  4512 non-null   object
 14  URL                     4512 non-null   object
dtypes: i

In [18]:
# Displaying how many rows(recipes) we've removed
print(c,"Recipes have been removed.")

2359 Recipes have been removed.


### A total of 2359 rows(recipes) have been removed. This can be verified by comparing the difference in number of rows from the original dataframe(6871) and filtered dataframe(4512) = 2359.

In [19]:
# Saving the new data into FilteredIndianFoodRecipes.csv
IndianRecipes.to_csv("IndianFoodDataset/FilteredIndianFoodRecipes.csv")

---

## Tracking nutrients

### Nutrients to be tracked -

- Carbs
- Proteins
- Fats
    - Saturated Fats
    - Unsaturated Fats

In [20]:
IndianRecipes_Nutrients = pd.read_csv('IndianFoodDataset/FilteredIndianFoodRecipes.csv')

In [21]:
#Adding Carbohydrates column
IndianRecipes_Nutrients.insert(16, "Carbohydrates",0)

In [22]:
#Adding Energy column
IndianRecipes_Nutrients.insert(17, "Energy",0)

In [23]:
#Adding Protien column
IndianRecipes_Nutrients.insert(18, "Protein",0)

In [24]:
#Adding Fat column
IndianRecipes_Nutrients.insert(19, "Fat",0)

In [25]:
#Adding Fibre column
IndianRecipes_Nutrients.insert(20, "Fibre",0)

In [26]:
IndianRecipes_Nutrients

Unnamed: 0   Srno                                         RecipeName  \
0              0      1                               Masala Karela Recipe   
1              1      2  टमाटर पुलियोगरे रेसिपी - Spicy Tomato Rice (Re...   
2              3      4  Gongura Chicken Curry Recipe - Andhra Style Go...   
3              4      5  आंध्रा स्टाइल आलम पचड़ी रेसिपी - Adrak Chutney ...   
4              5      6  Pudina Khara Pongal Recipe (Rice and Lentils C...   
...          ...    ...                                                ...   
4507        6865  14045        Quinoa Phirnee Recipe (Quinoa Milk Pudding)   
4508        6866  14073  गोअन मशरुम जकुटी रेसिपी - Goan Mushroom Xacuti...   
4509        6867  14107  शकरकंदी और मेथी का पराठा रेसिपी - Sweet Potato...   
4510        6868  14165      Ullikadala Pulusu Recipe | Spring Onion Curry   
4511        6870  14211              नवरंग दाल रेसिपी - Navrang Dal Recipe   

                                   TranslatedRecipeName  \
0                                  Masala Karela Recipe   
1                            Spicy Tomato Rice (Recipe)   
2     Gongura Chicken Curry Recipe - Andhra Style Go...   
3     Andhra Style Alam Pachadi Recipe - Adrak Chutn...   
4     Pudina Khara Pongal Recipe (Rice and Lentils C...   
...                                                 ...   
4507        Quinoa Phirnee Recipe (Quinoa Milk Pudding)   
4508                        Goan Mushroom Xacuti Recipe   
4509        Sweet Potato & Methi Stuffed Paratha Recipe   
4510      Ullikadala Pulusu Recipe | Spring Onion Curry   
4511                                 Navrang Dal Recipe   

                                            Ingredients  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...   
1     2-1/2 कप चावल - पका ले,3 टमाटर,3 छोटा चमच्च बी...   
2     500 grams Chicken,2 Onion - chopped,1 Tomato -...   
3     1 बड़ा चमच्च चना दाल,1 बड़ा चमच्च सफ़ेद उरद दाल,2...   
4     1 cup Rice - soaked for 20 minutes,1/2 cup Yel...   
...                                                 ...   
4507  1 cup Quinoa,3/4 cup Sugar,1 teaspoon Cardamom...   
4508  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...   
4509  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...   
4510  150 grams Spring Onion (Bulb & Greens) - chopp...   
4511  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...   

                                  TranslatedIngredients  PrepTimeInMins  \
0     6 Karela (Bitter Gourd/ Pavakkai) - deseeded,S...              15   
1     2-1 / 2 cups rice - cooked, 3 tomatoes, 3 teas...               5   
2     500 grams Chicken,2 Onion - chopped,1 Tomato -...              15   
3     1 tablespoon chana dal, 1 tablespoon white ura...              10   
4     1 cup Rice - soaked for 20 minutes,1/2 cup Yel...              10   
...                                                 ...             ...   
4507  1 cup Quinoa,3/4 cup Sugar,1 teaspoon Cardamom...              10   
4508  20 बटन मशरुम,2 प्याज - काट ले,1 टमाटर - बारीक ...              15   
4509  1 बड़ा चम्मच तेल,1 कप गेहूं का आटा,नमक - स्वाद ...              30   
4510  150 grams Spring Onion (Bulb & Greens) - chopp...               5   
4511  2 बड़े चम्मच हरी मूंग दाल,2 बड़े चम्मच सफ़ेद उरद ...              10   

      CookTimeInMins  TotalTimeInMins  Servings  ...                  Course  \
0                 30               45         6  ...               Side Dish   
1                 10               15         3  ...             Main Course   
2                 30               45         4  ...                   Lunch   
3                 20               30         4  ...  South Indian Breakfast   
4                 20               30         4  ...  South Indian Breakfast   
...              ...              ...       ...  ...                     ...   
4507              25               35         2  ...                 Dessert   
4508              45               60         4  ...                   Lunch   
4509              60          

In [27]:
table1

code                                             name  regn       water  \
0    A001       Amaranth seed, black (Amaranthus cruentus)     1        9.89   
1    A002  Amaranth seed, pale brown (Amaranthus cruentus)     6   9.20±0.40   
2    A003                    Bajra (Pennisetum typhoideum)     6   8.97±0.60   
3    A004                         Barley (Hordeum vulgare)     6   9.77±0.38   
4    A005                          Jowar (Sorghum vulgare)     6   9.01±0.77   
..    ...                                              ...   ...         ...   
523  S006                              Rohu (Labeo rohita)     6  76.34±0.28   
524  S007                          Crab (Pachygrapsus sp.)     1       80.58   
525  S008          Prawns, big (Macrobrachium rosenbergii)     1       77.43   
526  S009                Prawns, small (Macrobrachium sp.)     3       82.58   
527  S010                 Tiger prawns (Macrobrachium sp.)     2       83.24   

        protcnt        ash      fatce       fibtg     fibins     fibsol  \
0         14.59       2.78       5.74        7.02       5.76       1.26   
1    13.27±0.34  3.05±0.30  5.56±0.33   7.47±0.09  5.80±0.17  1.67±0.21   
2    10.96±0.26  1.37±0.17  5.43±0.64  11.49±0.62  9.14±0.58  2.34±0.42   
3    10.94±0.51  1.06±0.22  1.30±0.20  15.64±0.64  9.98±0.62  5.66±0.68   
4     9.97±0.43  1.39±0.34  1.73±0.31  10.22±0.49  8.49±0.40  1.73±0.40   
..          ...        ...        ...         ...        ...        ...   
523  19.71±0.57  1.28±0.04  2.39±0.64         NaN        NaN        NaN   
524       13.23       1.23       0.89         NaN        NaN        NaN   
525       19.24       0.84       0.52         NaN        NaN        NaN   
526       13.07       0.86       0.78         NaN        NaN        NaN   
527       14.24       0.83       0.66         NaN        NaN        NaN   

       choavldf    enerc  
0         59.98     1490  
1    61.46±0.60  1489±10  
2    61.78±0.85  1456±18  
3    61.29±0.77  1321±19  
4    67.68±1.03  1398±13  
..          ...      ...  
523         NaN   428±12  
524         NaN      327  
525         NaN      380  
526         NaN      297  
527         NaN      284  

[528 rows x 12 columns]